### DISCUSSION TITLE
Route parameter to model binding

### DISCUSSION THREAD

**Discussion by notramo** [luckyframework/lucky#1779] [Feature requests & Ideas]
Laravel route model binding is very useful in reducing boilerplate.
https://laravel.com/docs/9.x/routing#route-model-binding

How it would work in Lucky?
When a route parameter contains an `_id` suffix, automatically define a method, without the `_id` suffix that retrieves and memoize the given instance of a model. E.g. if the route param is `employee_id`, Lucky woud define the `employee` method that retrieves an `Employee` with the given id, and return 404 if not found. It could also check the relationships for nested resources.
```crystal
class Employee::Show < ApiAction
  get "/employees/:employee_id" do
    puts employee.class # => Employee
  end
end
```

**Comment by jwoertink** [comment]
This wouldn't really be possible in Lucky because you don't use the Model for doing queries. Since we use a separate object, there wouldn't be a way to tell which one you wanted without other boilerplate.

```crystal
class Employees::Show < ApiAction
  get "/employees/:employee_id" do
    # We can't just assume you would use EmployeeQuery here without side affects
    employee = UserQuery.new.employee_id(1234).first
   json(EmployeeSerializer.new(employee))
  end
end
```

Though, maybe this could be a good case for a separate shard? Then in that case there's at least an understanding of how things are done... Might be worth looking in to.

**Reply by matthewmcgarvey** [reply]
Definitely a separate shard. We just untied Lucky from Avram. I'd not support adding it back.
